### PR TITLE
Ns 315/finish implementing endpoints

### PR DESCRIPTION
**Pull request by net-illia-stetsenko**

### CODE REVIEW

File: net-gateway/Cargo.toml [hunk 1]
```diff
@@ -21,8 +21,9 @@ tokio = { version = "1.28.1", features = ["full"] }
 async-tungstenite = { version = "0.25.1", features = ["tokio-runtime"] }
 net-core-api = "0.5.0"
 net-agent-api = "0.1.5"
-net-reporter-api = "0.6.0"
-net-inserter-api = "0.2.0"
+net-reporter-api = { git = "https://github.com/net-stalker/net-api", branch = "NS-315/implement-api-for-network-requests" }
+net-inserter-api = { git = "https://github.com/net-stalker/net-api", branch = "NS-315/implement-api-for-network-requests" }
+net-updater-api = { git = "https://github.com/net-stalker/net-api", branch = "NS-315/implement-api-for-network-requests" }
```
Comment (net-yehor-tretiakov) [inline net-gateway/Cargo.toml]: Please, do not merge like this

File: net-gateway/src/endpoints/buffer/flush.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use net_token_verifier::fusion_auth::fusion_auth_verifier::FusionAuthVerifier;
+use net_updater_api::api::updaters::buffer_flush::buffer_flush_request::BufferFlushRequestDTO;
+use crate::core::quinn_client_endpoint_manager::QuinnClientEndpointManager;
+use crate::core::user_facing_error::UserFacingError;
+use crate::{authorization, config::Config};
+use net_core_api::core::typed_api::Typed;
+use net_core_api::core::encoder_api::Encoder;
+
+
+#[patch("/buffer")]
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/buffer/flush.rs:18]: It should be `flush` or `buffer_flush` I believe, isnt it? Sounds strange, just `buffer`

File: net-gateway/src/endpoints/buffer/flush.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        Ok(server_connection) => server_connection,
+        Err(_) => return Err(UserFacingError::Timeout),
+    };
+
+    let mut delete = false;
+    if let Some(item) = payload.next().await {
+        if item.unwrap().name() == "delete" {
+            delete = true;
+        }
+    }
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/buffer/flush.rs]: How exactly should it work?

File: net-gateway/src/endpoints/files/pcap_file.rs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+use net_core_api::core::encoder_api::Encoder;
+
+use net_inserter_api::api::pcap_file::InsertPcapFileDTO;
+use net_token_verifier::fusion_auth::fusion_auth_verifier::FusionAuthVerifier;
+
+use crate::core::quinn_client_endpoint_manager::QuinnClientEndpointManager;
+use crate::core::user_facing_error::UserFacingError;
+use crate::{authorization, config::Config};
+
+#[post("/pcap-file")]
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/files/pcap_file.rs]: Still voting for `insert` in the name somewhere. It is impossible to understand, what is going on if not diving into the code

File: net-gateway/src/endpoints/networks/add_network.rs
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/networks/add_network.rs:1]: This front namings..

File: net-gateway/src/endpoints/networks/add_network.rs [hunk 5]
```diff
@@ -66,35 +63,16 @@ async fn network(
 
     match server_connection.send_all_reliable(&request.encode()).await {
         Ok(_) => (),
-        Err(e) => return HttpResponse::InternalServerError().json(
-            serde_json::json!({
-                "error": "Internal Server Error",
-                "message": format!("Cound't insert the data: {}", e.to_string()),
-            })
-        ),
+        Err(_) => return Err(UserFacingError::Timeout),
```
Comment (net-yehor-tretiakov) [inline net-gateway/src/endpoints/networks/add_network.rs]: Should probably be `unwrap_or` here, instead of this. And, like... Everywhere

**Review by net-yehor-tretiakov — COMMENTED** [review]
Wanna talk about it

File: net-gateway/Cargo.toml (hunk 1 above)
Comment (net-illia-stetsenko) [reply net-gateway/Cargo.toml]: Of course

File: net-gateway/src/endpoints/files/pcap_file.rs (hunk 4 above)
Comment (net-illia-stetsenko) [reply net-gateway/src/endpoints/files/pcap_file.rs]: `We should not use URIs to indicate a CRUD function. URIs should only be used to identify the resources and not any action upon them uniquely.`. You should read it like post pcap-file

File: net-gateway/src/endpoints/buffer/flush.rs (hunk 2 above)
Comment (net-illia-stetsenko) [reply net-gateway/src/endpoints/buffer/flush.rs:18]: `We should not use URIs to indicate a CRUD function. URIs should only be used to identify the resources and not any action upon them uniquely.`. You should read it like patch buffer.

File: net-gateway/src/endpoints/buffer/flush.rs (hunk 3 above)
Comment (net-illia-stetsenko) [reply net-gateway/src/endpoints/buffer/flush.rs]: If there is a `delete` item in the payload then we should just clear the buffer and not to transfer the data into the main table

File: net-gateway/src/endpoints/networks/add_network.rs (hunk 5 above)
Comment (net-illia-stetsenko) [reply net-gateway/src/endpoints/networks/add_network.rs:66]: I will fix this later

Comment (net-illia-stetsenko) [reply net-gateway/src/endpoints/networks/add_network.rs]: We must return something from endpoint anyway

**Review by net-yehor-tretiakov — APPROVED** [review]

**Pull request merged**